webtest/app.py

```diff
@@ -204,8 +205,17 @@ def set_authorization(self, value):
                     val = ':'.join(list(val))
                     val = b64encode(to_bytes(val)).strip()
                     val = val.decode('latin1')
-                elif authtype == 'Bearer' and val and isinstance(val, (str, unicode)):
-                    val = val.strip()
+                elif authtype == 'Bearer':
+                    if sys.version_info >= (3, 0, 0):
```

Comment:
You've added too much complexity. Testing isinstance(val, (str, text_type)) should work with both py2 and py3